kratos/processes/apply_component_table_process.hpp

```diff
@@ -0,0 +1,196 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:		 BSD License
+//					 Kratos default license: kratos/license.txt
+//
+//  Main authors:    Ignasi de Pouplana
+//  Collaborator:    Alejandro Cornejo
+//
+
+#if !defined(KRATOS_APPLY_COMPONENT_TABLE_PROCESS)
+#define  KRATOS_APPLY_COMPONENT_TABLE_PROCESS
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "includes/table.h"
+#include "includes/kratos_flags.h"
+#include "includes/kratos_parameters.h"
+#include "processes/process.h"
+
+namespace Kratos
+{
+
+///@name Kratos Classes
+///@{
+
+/// The base class for all processes in Kratos.
+/** This function applies a table value to a component
+*/
+class ApplyComponentTableProcess : public Process
+{
+    
+public:
+    ///@name Type Definitions
+    ///@{
+
+    /// Pointer definition of ApplyConstantScalarValueProcess
+    KRATOS_CLASS_POINTER_DEFINITION(ApplyComponentTableProcess);
+    
+    /// Defining a table with double argument and result type as table type.
+    typedef Table<double,double> TableType;
+    
+    /// Constructor
+    ApplyComponentTableProcess(ModelPart& model_part,
```

Comment:
this name is not really descriptive
Please use sth like ApplyValueFromTableToNodesProcess or sth similar to clarify what actually is going on